### PR TITLE
Added names as unique identifiers to courses

### PR DESCRIPTION
**Pull request by JakeChvatal**
As discussed in the issue.

To test: ```yarn test```

Also ensure that the produced audit visible in the test case uses the desired convention for names and that the 'no duplicates' heuristic is now properly respected.

### CODE REVIEW

File: test/html_parser.test.js [hunk 1]
```diff
@@ -710,6 +713,365 @@ test('Verifies that the CS degree audit is properly reproduced by the code', ()
...
+                        classId: '4500' 
+                    },
+                    { 
+                        subject: 'THTR', 
+                        classId: '1170' 
+                    },
+                    { 
+                        subject: 'CS', 
+                        list: [ '4100', '4300', '4410','4150','4550','4991','4900'],
+                        num_required: '1'
```
Comment (bugsalexander) [inline test/html_parser.test.js]: It would be cool if we had a "type" property here for the type of requirement needed. I'll open another issue. 

Ex. this one might have type "or" or "union", for ranges we could have type "range"

File: test/html_parser.test.js [hunk 2]
```diff
@@ -1225,6 +1339,7 @@ test('Verifies that the CS Math degree audit is properly reproduced by the code'
                         "hon":false,
                         "subject":"ARTS",
                         "classId":"1990",
+                        "name": "Ceramics:WheelThrowing",
```
Comment (bugsalexander) [inline test/html_parser.test.js]: Wheel Throwing LOL

Comment (JakeChvatal) [reply test/html_parser.test.js]: There's a harry potter course!

File: test/html_parser.test.js (hunk 1 above)
Comment (JakeChvatal) [reply test/html_parser.test.js]: I have another issue out for that - figure it should have been distinct.
